### PR TITLE
Setup JIRA sync for Community Issues and PRs via Github actions

### PR DESCRIPTION
**Pull request by david-yu**
### Description

- Sets up sync to JIRA for community issues to JIRA regardless of who files them (team member or not)
- Sets up sync to JIRA for community PRs to JIRA if non-team members file PR to track for review.  
 
### Testing & Reproduction steps
* In the case of bugs, describe how to replicate
* If any manual tests were done, document the steps and the conditions to replicate
* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding

### Links
Include any links here that might be helpful for people reviewing your PR (Tickets, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.

Please be mindful not to leak any customer or confidential information. HashiCorp employees may want to use our internal URL shortener to obfuscate links.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

### CODE REVIEW

**Comment by dhirennotani19** [comment]
Hi @david-yu 

This is Dhiren from Team Exalate.

[Exalate](https://marketplace.atlassian.com/apps/1213645/exalate-jira-issue-sync-more?hosting=cloud&tab=overview), provides a fully bi-directional synchronization between GitHub issues and Jira issues (among other ITSM systems) that is fully customizable. It is easy to set up and Exalate uses a scripting engine which is based on Groovy scripts where you can control what data you want to share and even you can independently handle the incoming information. You can achieve complex use-cases with ease. 

You can easily fetch data from GitHub repositories and sync issues from GitHub to issues in Jira.

If you would like to see a customized demo of the product in action, feel free to [book a slot](https://exalate.com/book-demo/) with us. 

Thanks
Dhiren

**Review by curtbushko — APPROVED** [review]
Looks good!

File: .github/workflows/jira.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            echo "::set-output name=message::true"
+          else
+            echo "Actor ${{ github.actor }} is NOT a ${TEAM} team member"
+            echo "::set-output name=message::false"
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.JIRA_SYNC_GITHUB_TOKEN }}
+
+      - name: Create ticket
+        if: ( github.event.action == 'opened' && steps.set-ticket-type.outputs.type != 'Task' ) || ( github.event.action == 'opened' && steps.set-ticket-type.outputs.type == 'Task' && steps.is-team-member.outputs.message == 'false' )
```
Comment (wilkermichael) [inline .github/workflows/jira.yaml:52]: Just want to make sure I understand the intent behind `is-team-member`. This keeps JIRA from auto generating tickets we create in GitHub right?

**Review by wilkermichael — APPROVED** [review]

File: .github/workflows/jira.yaml (hunk 1 above)
Comment (david-yu) [reply .github/workflows/jira.yaml:52]: Yes, that is correct, this is mostly a copy and pasted from what's in https://github.com/hashicorp/hcdiag/blob/main/.github/workflows/jira.yml

Comment (wilkermichael) [reply .github/workflows/jira.yaml:52]: ok cool, all good!

**Pull request merged**